### DISCUSSION TITLE
ctx.command object guaranteed to be identical to attributes of the cog class?

### DISCUSSION THREAD

**Discussion by ioistired** [Rapptz/discord.py#5864] [Q&A] [answered]
```py
class Foo(commands.Cog):
    cmds = set()
    def deco(cmd, cmds=cmds):
        cmds.add(cmd)
        return cmd

    def cog_check(self, ctx):
        return ctx.command in self.cmds

    @deco
    @commands.command()
    async def allowed(self, ctx):
        await ctx.send('allowed')

    @commands.command()
    async def denied(self, ctx):
        await ctx.send('illegal')
```
neither of these commands run. why is that?

**Comment by ioistired** [comment]
For now I will use [`Command.qualified_name`](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.Command.qualified_name) instead since that is globally-unique. Not sure if that's the correct solution, though.

**Comment by Rapptz** [comment, marked as answer]
Commands in cogs undergo copy operations to allow multiple cog instances to have separate command instances.

Comparison between instances is generally not allowed (evident by the fact there's no `__eq__` or even `__hash__`) -- using `Command.qualified_name` is the proper solution here.
